doc/build/kconfig/setting.rst

```diff
@@ -15,6 +15,37 @@ All Kconfig options can be searched in the :ref:`Kconfig search page
    Before making changes to Kconfig files, it's a good idea to also go through
    the :ref:`kconfig_tips_and_tricks` page.
 
+.. _configuration_file_types:
+
+Kconfig configuration file types
+********************************
+
+There are two types of Kconfig configuration files:
+
+1. **Kconfig Files**: These files use the following syntax:
+
+   .. code-block:: none
+
+      config CPU_HAS_FPU
+      bool
+      help
+         This symbol is ``y`` if the CPU has a hardware floating point unit.
```

Comment:
this is a pure duplicate of the below example:
https://github.com/zephyrproject-rtos/zephyr/blob/18b18713515cdf2ad5783a19b1b9905690be5b53/doc/build/kconfig/setting.rst?plain=1#L83-L86

If this should be an introduction to Kconfig syntax / an example, then I think it would be much better to describe exactly what each field means, perhaps like:
```
config <symbol>
    < <type> | <type> "prompt" [if <another_symbol>] >
    [select <symbol> [if <another_symbol>]]
    [depends on <another_symbol>]
    help
      <help text>
```
that can help understanding the use of the terms, `prompt`, `symbol`, and give an introduction to valid type, like `bool`, `string`, `int`, `hex`, etc.

Although we should be careful not to just repeat what is already well-described in the official Kconfig language guide: https://docs.kernel.org/kbuild/kconfig-language.html#kconfig-syntax